### DISCUSSION TITLE
Not Found 404 (in streamed page)

### DISCUSSION THREAD

**Discussion by votemike** [vercel/next.js#62090] [Help]
### Summary

I would like to gather best practice opinions about status codes for not-found products on an ecommerce website.

### Additional information

I am building an ecommerce website. Products and categories may come and go. I would like URLs for products/categories that no longer exist to 404.
The layout for the page is streamed in order to show the user something as fast as possible.
However, streamed pages where the content isn't found will still 200: https://nextjs.org/docs/app/api-reference/file-conventions/not-found
Vercel demo this here: https://demo.vercel.store/product/non-existent
While the page says 404, the status code is 200.
My understanding is that SEO best practice is to 404 instead of 200.
Does anyone have an suggestion on the best thing to do?

### Example

https://demo.vercel.store/product/non-existent

**Comment by tiacop** [comment]
If you have a loading.tsx remove it. not-found.tsx will return 404 if no loading.tsx is included.
see:
https://github.com/vercel/next.js/discussions/64500#discussioncomment-9123027

**Reply by votemike** [reply]
Does the demo vercel store have a loading.tsx?
I would have thought that the Vercel demo would be the ideal way to set up an e-commerce store. And if a non-existent page is not 404ing, then that's not ideal.
